client/verta/verta/client.py

```diff
@@ -2038,16 +2045,77 @@ def _log_artifact(self, key, artifact, artifact_type, extension=None, method=Non
             else:
                 _utils.raise_for_http_error(response)
 
-        # upload artifact to artifact store
-        url = self._get_url_for_artifact(key, "PUT")
-        artifact_stream.seek(0)  # reuse stream that was created for checksum
+        self._upload_artifact(key, artifact_stream)
+
+    def _upload_artifact(self, key, artifact_stream, part_size=64*(10**6)):
+        """
+        Uploads `artifact_stream` to ModelDB artifact store.
+
+        Parameters
+        ----------
+        key : str
+        artifact_stream : file-like
+        part_size : int, default 64*(10**6)
+            If using multipart upload, number of bytes to upload per part.
+
+        """
+        artifact_stream.seek(0)
         if self._conf.debug:
-            print("[DEBUG] uploading {} bytes ({})".format(len(artifact_stream.read()), basename))
+            print("[DEBUG] uploading {} bytes ({})".format(len(artifact_stream.read()), key))
             artifact_stream.seek(0)
 
-        response = _utils.make_request("PUT", url, self._conn, data=artifact_stream)
-        _utils.raise_for_http_error(response)
-        print("upload complete ({})".format(basename))
+        # check if multipart upload ok
+        try:
+            url_for_artifact = self._get_url_for_artifact(key, "PUT", part_num=1)
+        except requests.HTTPError:  # backend knows it doesn't support multipart upload, e.g. NFS
```

Comment:
It shouldn't raise an exception. It should handle it gracefully. We added a flag to the protos specifically for this.

Reply:
Makes sense to me. I'll check on that in the backend